core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2044,16 +2044,20 @@ public void delete(AlluxioURI path, DeleteContext context)
           createLockingScheme(path, context.getOptions().getCommonOptions(),
               LockPattern.WRITE_EDGE);
       try (LockedInodePath inodePath = mInodeTree
-              .lockInodePath(lockingScheme, rpcContext.getJournalContext())
-      ) {
-        mPermissionChecker.checkParentPermission(Mode.Bits.WRITE, inodePath);
-
+              .lockInodePath(lockingScheme, rpcContext.getJournalContext())) {
         // If the mount point is read only, we allow removing the in-Alluxio metadata and data
         // in order to load it from the UFS again.
         // This can happen if Alluxio is out-of-sync with the UFS.
         if (!context.getOptions().getAlluxioOnly()) {
           mMountTable.checkUnderWritableMountPoint(path);
         }
+
+        mPermissionChecker.checkParentPermission(Mode.Bits.WRITE_EXECUTE, inodePath);
```

Comment:
deletion needs the write_execute permission to the object itself? Or just requires the permission of its parent?

Reply:
Deletion does not require the permission of the object itself.

Reply:
could you please also help check whether the ranger is following the same rule? 
Need to make sure this change doesn't regress the other logic.